proto/frequenz/api/reporting/v1/reporting.proto

```diff
@@ -0,0 +1,538 @@
+  // Encapsulates the result of aggregating a metric.
+  message AggregatedResult {
+    // Metric and related formula provided for aggregation.
+    AggregationConfig aggregation_config = 1;
+
+    // A list of aggregated metrics.
+    repeated SimpleAggregatedMetricSample samples = 2;
+  }
+
+  // List of aggregated results, each corresponding to a metric and custom aggregation
+  // formula.
+  //
+  // !!! note
+  //     Each entry in this list contains the aggregation formula config and the
+  //     corresponding aggregated metric samples for the requested timeframe.
+  repeated AggregatedResult results = 1;
+
+  // Metadata for pagination, containing the token for the next page of results.
+  //
+  // !!! note
+  //     If `pagination_info` is populated, it implies that more data is available to fetch.
+  frequenz.api.common.v1.pagination.PaginationInfo pagination_info = 2;
+}
+
+// Message defining the request format for streaming aggregated historical metrics.
+// This request allows to specify custom aggregation formulas, along with general
+// filtering.
+//
+// At least one aggregation formula config must be provided. The aggregation
+// follows the passive sign convention.
+//
+message ReceiveAggregatedMicrogridComponentsDataStreamRequest {
+  // General filter criteria for querying microgrid components data.
+  message AggregationStreamFilter {
+    // Optional resampling options like resolution for the data, represented in seconds.
+    // If omitted, data will be returned in its original representation.
+    ResamplingOptions resampling_options = 1;
+  }
+
+  // List of pairs of metric and corresponding aggregation formula.
+  repeated AggregationConfig aggregation_configs = 1;
+
+  // General streaming filter that applies to all formula aggregations.
+  AggregationStreamFilter filter = 2;
+}
+
+// Message defining the response format for a stream that fetches aggregated real-time metrics
+// for the provided custom aggregation formulas.
+//
+// !!! note
+//     The formula and metric must have been specified in the corresponding request.
+//     A single aggregated sample for the metric is returned in the sample field. Each message
+//     covers a single formula. For multiple formulars provided in the request, expect sequential
+//     messages in the stream.
+//
+// !!! example
+//     Given a stream output, a single sample might be:
+//     ```
+//     {
+//       aggregation_config: {
+//         microgrid_id: 1,
+//         metric: "DC_VOLTAGE_V",
+//         aggregation_formula: "avg(1,2,3)"
+//       },
+//       sample {
+//         sampled_at: '2023-10-01T00:00:00Z',
+//         sample: { value: 42.5 }
+//       }
+//     }
+//     ```
+//
+message ReceiveAggregatedMicrogridComponentsDataStreamResponse {
+  // The metric and formula that has been used to aggregate the sample.
+  AggregationConfig aggregation_config = 1;
+
+  // Aggregated sample value and corresponding UTC timestamp when it was generated.
```

Comment:
@cwasicki FYI re: the metric sample timestamp, this comment should probably be updated as well. Just a note to myself mainly.